src/index.ts

```diff
@@ -62,18 +62,25 @@ export type ImgixUrlQueryParams = {
     faceindex?: number;
     facepad?: number;
 };
-
-const pickTrueInObject = <K extends string>(obj: Record<K, boolean>) =>
-    pickBy(obj, (_key, value): value is true => value);
+// TODO: remove cast and either make all functions allow partial, or all non-partial.
+// const pickTrueInObject = <K extends string>(obj: Record<K, boolean>) =>
+const pickTrueInObject = <K extends string>(
+    obj: Partial<Record<K, boolean>>,
+): Partial<Record<K, true>> =>
+    pickBy(obj as Record<K, boolean>, (_key, value): value is true => value);
```

Comment:
I'm a bit stuck here.

`pickTrueInObject` must be allowed to receive partials. But if we update this function, we have to update every helper it uses internally (`pickBy` and `entries`) to also allow `Partial`. That feels wrong :-/

@karol-majewski Any thoughts on what we should do here?

Reply:
@karol-majewski Hey, do you have any thoughts on the above?

Reply:
If the `T` passed to `pickTrueInObject` is a `Dictionary<boolean>`, isn't it sufficient to say `pickTrueInObject` will return a `Partial<T>`?

Sure, the signature could be more precise:

```ts
declare function pickTrueInObject<T extends object>(source: T): Record<PropertyOfValue<T, boolean>, true>;

type PropertyOfValue<T, V> = {
    [K in keyof T]: T[K] extends V
      ? K
      : never
  }[keyof T];


declare const foo: {
  foo: number;
  bar: boolean;
}

let result = pickTrueInObject(foo); // { bar: true }
```

but since we're not dealing with object literals, we cannot use the type system to tell us which properties will be `true` in runtime and that renders having this extra information useless.

Reply:
This seems to pipe well.

```ts
import { pipe } from 'pipe-ts';

declare function pickTrueInObject<T extends object>(source: T): Record<PropertyOfValue<T, boolean>, true>;

type PropertyOfValue<T, V> = {
    [K in keyof T]: T[K] extends V
      ? K
      : never
  }[keyof T];


declare const foo: {
  foo: number;
  bar: boolean;
}

const UNSAFE_keys = <T extends object>(source: T) =>
  Object.keys(source) as (keyof T)[];

const keys = pipe(pickTrueInObject, UNSAFE_keys)(foo); // "bar"[]
```